<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>UdpDemoClass</class>
 <widget class="QMainWindow" name="UdpDemoClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>840</width>
    <height>357</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>UdpDemo</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>811</width>
      <height>307</height>
     </rect>
    </property>
    <property name="title">
     <string/>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QFrame" name="frame">
       <property name="frameShape">
        <enum>QFrame::Panel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>绑定端口</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="sbBindPort">
          <property name="minimumSize">
           <size>
            <width>50</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximum">
           <number>100000</number>
          </property>
          <property name="value">
           <number>1200</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>目标地址</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="cbHostAddr">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>100</width>
            <height>0</height>
           </size>
          </property>
          <property name="editable">
           <bool>true</bool>
          </property>
          <item>
           <property name="text">
            <string>127.0.0.1</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>目标端口</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="sbHostPort">
          <property name="minimumSize">
           <size>
            <width>50</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximum">
           <number>100000</number>
          </property>
          <property name="value">
           <number>1201</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>组播地址</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="cbMulticastAddr">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>100</width>
            <height>0</height>
           </size>
          </property>
          <property name="editable">
           <bool>true</bool>
          </property>
          <property name="currentText">
           <string>239.0.0.100</string>
          </property>
          <item>
           <property name="text">
            <string>127.0.0.1</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>组播端口</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="sbMulticastPort">
          <property name="minimumSize">
           <size>
            <width>50</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximum">
           <number>100000</number>
          </property>
          <property name="value">
           <number>35566</number>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QFrame" name="frame_2">
       <property name="frameShape">
        <enum>QFrame::Panel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>消息内容</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="txtMsg">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>300</width>
            <height>0</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnSendMsg">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>发送消息</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnBroadCast">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>广播消息</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnMulticast">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>组播消息</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QListWidget" name="lstMsg"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextUnderIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actBind"/>
   <addaction name="actUnbind"/>
   <addaction name="separator"/>
   <addaction name="actJoinGroup"/>
   <addaction name="actLeaveGroup"/>
   <addaction name="separator"/>
   <addaction name="actClear"/>
   <addaction name="separator"/>
   <addaction name="actQuit"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actBind">
   <property name="icon">
    <iconset resource="UdpDemo.qrc">
     <normaloff>:/UdpDemo/Resources/APPTL.ICO</normaloff>:/UdpDemo/Resources/APPTL.ICO</iconset>
   </property>
   <property name="text">
    <string>绑定端口</string>
   </property>
  </action>
  <action name="actUnbind">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="UdpDemo.qrc">
     <normaloff>:/UdpDemo/Resources/ACTIVITL.ICO</normaloff>:/UdpDemo/Resources/ACTIVITL.ICO</iconset>
   </property>
   <property name="text">
    <string>解绑端口</string>
   </property>
  </action>
  <action name="actClear">
   <property name="icon">
    <iconset resource="UdpDemo.qrc">
     <normaloff>:/UdpDemo/Resources/CNFNOT.ICO</normaloff>:/UdpDemo/Resources/CNFNOT.ICO</iconset>
   </property>
   <property name="text">
    <string>清空报文</string>
   </property>
  </action>
  <action name="actQuit">
   <property name="icon">
    <iconset resource="UdpDemo.qrc">
     <normaloff>:/UdpDemo/Resources/OOFL.ICO</normaloff>:/UdpDemo/Resources/OOFL.ICO</iconset>
   </property>
   <property name="text">
    <string>退出程序</string>
   </property>
  </action>
  <action name="actJoinGroup">
   <property name="icon">
    <iconset resource="UdpDemo.qrc">
     <normaloff>:/UdpDemo/Resources/FOLDER.ICO</normaloff>:/UdpDemo/Resources/FOLDER.ICO</iconset>
   </property>
   <property name="text">
    <string>加入组播</string>
   </property>
  </action>
  <action name="actLeaveGroup">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="UdpDemo.qrc">
     <normaloff>:/UdpDemo/Resources/MYSL.ICO</normaloff>:/UdpDemo/Resources/MYSL.ICO</iconset>
   </property>
   <property name="text">
    <string>退出组播</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="UdpDemo.qrc"/>
 </resources>
 <connections/>
</ui>
